### PR TITLE
IID tests improvements

### PR DESCRIPTION
**Pull request by maksymmalyhin**
A bit nicer replacement for #3065

### CODE REVIEW

**Comment by paulb777** [comment]
Thanks @maksymmalyhin 

While you're in the test, please fix the spelling of completionExpectaion to completionExpectation

File: Example/InstanceID/Tests/FIRInstanceIDKeyPairMigrationTest.m [hunk 1]
```diff
@@ -167,18 +161,15 @@ - (void)testUpdateKeyRefWithTagRetainsAndReleasesKeyRef {
...
-  // 2 = 1 from CFRetain() + 1 retained by `updateKeyRef`
-  XCTAssertEqual(CFGetRetainCount(publicKeyRef), 2);
+  // Should be still alive until execution finished
+  XCTAssertNotNil(publicKeyRef);
 
   [self waitForExpectationsWithTimeout:0.5 handler:NULL];
 
-  // No one else owns publicKeyRef except the test
-  XCTAssertEqual(CFGetRetainCount(publicKeyRef), 1);
+  // Should be release once finished
```
Comment (paulb777) [inline Example/InstanceID/Tests/FIRInstanceIDKeyPairMigrationTest.m]: s/release/released/

**Comment by maksymmalyhin** [comment]
@paulb777 I've added a couple more improvements that hopefully will improve reliability a bit. I would proceed with the changes on Travis green and keep improving tests as we go.

**Comment by paulb777** [comment]
Thanks @maksymmalyhin Looks like the verbose logging is showing multiple crashes in FIRInstanceIDTokenManagerTest. cc: @chliangGoogle

**Pull request merged**